Reference/Searching/Examine/Indexing/index.md

```diff
@@ -22,212 +13,123 @@ You can modify the built in indexes in several ways:
-
-There is some documentation about this in the [Examine documentation](https://shazwazza.github.io/Examine/configuration).
-
-The easiest way to modify how a field is configured is using a custom component, for example:
+## Creating a ConfigureOptions class
 
+We will start by creating a ConfigureExamineOptions class, that derives from `IConfigureNamedOptions<LuceneDirectoryIndexOptions>`:
 ```c#
-using Examine;
-using Umbraco.Core;
-using Umbraco.Core.Composing;
-
-// register the component
-public class CustomizeIndexComposer : ComponentComposer<CustomizeIndexComponent> { }
+using Examine.Lucene;
+using Microsoft.Extensions.Options;
 
-public class CustomizeIndexComponent : IComponent
+namespace MySite
 {
-    private readonly IExamineManager _examineManager;
-
-    public CustomizeIndexComponent(IExamineManager examineManager)
+    public class ConfigureExamineOptions : IConfigureNamedOptions<LuceneDirectoryIndexOptions>
     {
-        _examineManager = examineManager;
-    }
-
-    public void Initialize()
-    {
-        // get the external index
-        if (!_examineManager.TryGetIndex(UmbracoIndexes.ExternalIndexName, out IIndex index))
-            return;
-
-        // add a custom field type
-        index.FieldDefinitionCollection.TryAdd(new FieldDefinition("price", FieldDefinitionTypes.Double));
-
-        // modify an existing field type (not recommended)
-        index.FieldDefinitionCollection.AddOrUpdate(new FieldDefinition("parentID", FieldDefinitionTypes.FullText));
-    }
+        public void Configure(string name, LuceneDirectoryIndexOptions options)
+        {
+            throw new System.NotImplementedException();
+        }
 
-    public void Terminate()
-    {
+        public void Configure(LuceneDirectoryIndexOptions options)
+        {
+            throw new System.NotImplementedException();
+        }
     }
 }
 ```
 
-## Changing IValueSetValidator
-
-An `IValueSetValidator` is responsible for validating a `ValueSet` to see if it should be included in the index and/or filtering the data in the `ValueSet`. For example, by default the validation process for the ExternalIndex checks if a `ValueSet` has a category type of either "media" or "content" (not member). If a `ValueSet` was passed to the ExternalIndex and it did not pass this requirement it would be ignored.
+:::note
+In this documentation we will use the `ConfigureExamineOptions` class for everything, it is recommended that you seperate this class out per filter for simplicity, for example if you do a lot of configuring for the ExternalIndex class, you would have a `ConfigureExternalIndex` class
```

Comment:
```suggestion
This sample focuses solely on altering the external index and thus we name the class `ConfigureExternalIndexOptions`. If you are altering multiple indexes, it is recommended to have separate classes for each index - i.e. `ConfigureExternalIndexOptions` for the external index, `ConfigureInternalIndexOptions` for the internal index and so on.
```

Reply:
I figure we might as well document what we preach as recommended practice by naming the sample configuration appropriately?